File: app/src/main/java/com/example/sy43/db/DAO/CategoryDAO.java
package com.example.sy43.db.DAO;

import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Dao;
import androidx.room.Update;

import com.example.sy43.db.entity.Categorydb;
import com.google.common.util.concurrent.ListenableFuture;

import java.util.ArrayList;
import java.util.List;


@Dao
public
interface CategoryDAO{
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public ListenableFuture<Void> insert(Categorydb... categories);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    public ListenableFuture<Void> update(Categorydb... categories);


    @Query("SELECT * FROM Categorydb")
    public ListenableFuture<List<Categorydb>> findAll();

    @Query("SELECT * FROM Categorydb WHERE CatID LIKE :CatID")
    public ListenableFuture<Categorydb> findByID(int CatID);

    @Query("SELECT CurrentValue FROM Categorydb")
    public ListenableFuture<List<Float>> findValue();

    @Query("SELECT MaxValue FROM Categorydb")
    public ListenableFuture<List<Float>> findMaxValue();

    @Query("SELECT * FROM Categorydb WHERE isObjective = 1")
    public ListenableFuture<List<Categorydb>> findObjectives();

    @Query("SELECT * FROM Categorydb WHERE isObjective = 0")
    public ListenableFuture<List<Categorydb>> findCategories();

    @Delete
    public ListenableFuture<Integer>
    deleteCategory(List<Categorydb> category);
}

File: BokuDarjan/app/src/main/java/com/example/bokudarjan/category/CategoryDAO.kt
package com.example.bokudarjan.category

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.bokudarjan.bmonth.BMonth
import com.example.bokudarjan.category.Category

@Dao
/**
 * DAO Interface of [Category]
 */
interface CategoryDAO {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun addCategory(Category: Category);

    @Query("SELECT * FROM category_table ORDER BY categoryName")
    fun readAllData(): LiveData<List<Category>>;



    @Query("SELECT * FROM category_table WHERE categoryName = :name")
    fun getCategory(name: String): LiveData<List<Category>>

}